### PR TITLE
Create note with link to git-rebase.io

### LINKED ISSUE #1247: Add git rebase -i [SHA1] example.
**Which version of the book is affected?**

Problem is present in source and online books.

**Describe the bug:**

The book does not explain how to select the correct SHA1 commit to start a rebase operation on. The book has no example for the ```git rebase -i [sha1 of specific commit]``` command.

**Steps to reproduce:**

1. Go to https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History or https://git-scm.com/book/en/v2/Git-Branching-Rebasing.
1. Neither of them mention the ```git rebase -i [sha1 of specific commit]``` command.
1. Reader is not sure how to select the correct commit hash to start rebasing on.

**Expected behavior:**

1. Book contains a section on the ```git rebase -i [sha1 of specific commit]``` command.
1. Book explains how to select the correct starting hash to rebase on.
1. Book gives a guided example like the ones used for ```git rebase -i HEAD~3```.
1. Book explains methodology behing selecting the correct commit to rebase from.

**Screenshots:**

**Additional context:**

The git documentation(https://git-scm.com/docs/git-rebase#_interactive_mode) has some guidance:
```
Start it with the last commit you want to retain as-is:

git rebase -i <after-this-commit>
```

**Desktop:**

- Operating system: Ubuntu 18.04.2 LTS
- Browser/application: Firefox
- Browser/application version: 66.0.2

### LINKED ISSUE #1329: Link to, or incorporate (parts) of website https://git-rebase.io/
**Which version of the book is affected?**

Problem is present in source files.

**Describe the bug:**

I think the guidance on `git rebase` in the Pro Git book is not enough to actually learn how to use `git rebase` to perform the common operations like squashing commits, rebasing from upstream branch, reordering commits, fixing up unneeded commits.

**Steps to reproduce:**

This is a bit cheeky, so don't take this too seriously please. 😸
1. Try to learn how to use `git rebase` from Pro Git book.
2. Fail.
3. Read https://git-rebase.io/
4. Succeed in using `git rebase`.

**Expected behavior:**

After reading the parts: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History and https://git-scm.com/book/en/v2/Git-Branching-Rebasing the reader should be able to actually perform the mentioned commands for themselves in a real workflow.

**Screenshots:**

**Additional context:**

I found the website https://git-rebase.io/ to be extremely helpful to practice with `git rebase`. The examples are well written, easy to follow.

I think the Pro Git book would benefit massively from a guided "sandbox" tutorial on rewriting history.

I think the range of options are:
- Ask permission to link to the website.
- Ask permission to write our own examples, based on the work on the webpage.
- Ask permission to wholesale incorporate parts/the whole text in the book.
- Do a "clean room design" by letting someone write a tutorial on rewriting Git history, without ever having looked at the linked website.

I'll leave it up to the Pro Git maintainers to decide what/if action should be taken.

Related issue: #1247

### PR DESCRIPTION
**Pull request by HonkingGoose**
## What happens when merging this pull-request:

- Create a new section in the book with a link to git-rebase.io
- Closes #1247
- Closes #1329

## Pros and cons of proposed change:

Pro:
- We skip the work of creating a new tutorial/sandbox for `git rebase`.
- No maintenance costs for maintaining a separate guide.
- The linked guide contains instructions on setting up a 'sandbox environment', where users can just noodle around and learn how `git rebase` works in depth.
- I think that after reading the guide, users will be equipped to use `git rebase` in their own workflow effectively.

Con:
- We have no control over the editorial tone of the guide.
- The site might go down/go out of date.
- Content of the guide is not saved in our own repository.

## Discussion:

I've had a lot of time to learn and use `git rebase`, and have opened some (lengthy) issues in the past (#1247 and #1329) on how to improve the ProGit2 teaching material for `git rebase`.
I've come to the conclusion that we could just link to the great guide made by Drew DeVault on git-rebase.io and be done with this annoying HonkingGoose :smile: :sparkle:

### CODE REVIEW

File: book/07-git-tools/sections/rewriting-history.asc [hunk 1]
```diff
@@ -299,6 +299,16 @@ This changes the SHA-1s of the three most recent commits in your list, so make s
 Notice that the last commit (`f7f3f6d`) in the list is unchanged.
 Despite this commit being shown in the script, because it was marked as ``pick'' and was applied prior to any rebase changes, Git leaves the commit unmodified.
 
+[[_practical_exercises_for_git_rebase ]]
+==== Practical exercises for git rebase
```
Comment (ben) [inline book/07-git-tools/sections/rewriting-history.asc:303]: I don't think having an entire section just for a `NOTE` makes sense. Let's drop these few lines.

File: book/07-git-tools/sections/rewriting-history.asc [hunk 2]
```diff
@@ -299,6 +299,16 @@ This changes the SHA-1s of the three most recent commits in your list, so make s
...
+
+[NOTE]
+====
+Drew DeVault made a practical hands-on guide with exercises to learn how to use `git rebase`.
+You can find it at:
+https://git-rebase.io/
```
Comment (ben) [inline book/07-git-tools/sections/rewriting-history.asc:309]: What about this? That way it'll be linkified.

```suggestion
You can find it at https://git-rebase.io[].
```

File: book/07-git-tools/sections/rewriting-history.asc (hunk 1 above)
Comment (HonkingGoose) [reply book/07-git-tools/sections/rewriting-history.asc:303]: Okay, I'll drop the header and chapter reference then!

File: book/07-git-tools/sections/rewriting-history.asc (hunk 2 above)
Comment (HonkingGoose) [reply book/07-git-tools/sections/rewriting-history.asc:309]: Whoops, I forgot the `[]`, will go do that now. 😄

**Comment by HonkingGoose** [comment]
> Just a couple of formatting fixes, then we'll be good to go.

I've made the formatting changes you requested. Also I moved the link up one line, so that the link is right after the introductory text.

It's ready for another review. :+1:

**Comment by ben** [comment]
✨

**Comment by HonkingGoose** [comment]
Thanks for merging my work in! ✨

**Pull request merged**